### PR TITLE
[TRA-221] add OwnerShares and AllVaults query and cli

### PR DESCRIPTION
**Pull request by tqin7**
### Changelist
1. OwnerShares query: list all owner shares of a vault (paginated)
2. AllVaults query: list all vaults (paginated)
3. add CLI for these 2 queries

### Test Plan
1. unit tests
2. local manual testing

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Introduced new functionalities to query all vaults and owner shares within the vault system, enhancing user capabilities to interact with and manage vault data.
	- Added commands for listing all vaults and listing owner shares of a vault, improving navigation and data accessibility for users.

### CODE REVIEW

File: protocol/x/vault/keeper/grpc_query_shares_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		"Error: nil request": {
+			req:     nil,
+			vaultId: constants.Vault_Clob_0,
+			ownerShares: map[string]*big.Int{
+				constants.Alice_Num0.Owner: big.NewInt(100),
+				constants.Bob_Num0.Owner:   big.NewInt(200),
+			},
+			expectedErr: "invalid request",
+		},
+	}
```
Comment (Christopher-Li) [inline protocol/x/vault/keeper/grpc_query_shares_test.go:75]: What happens if a user sends a valid query, but paginates beyond the max size? 

Are there also any race conditions? For example the owners are constantly joining and leaving the vault, could a user of the query ever get an accurate full picture of all owners for a vault?

File: protocol/x/vault/keeper/grpc_query_shares.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		Type:   req.Type,
+		Number: req.Number,
+	}
+	_, exists := k.GetTotalShares(ctx, vaultId)
+	if !exists {
+		return nil, status.Error(codes.NotFound, "vault not found")
+	}
+
+	var ownerShares []*types.OwnerShare
+	ownerSharesStore := k.getVaultOwnerSharesStore(ctx, vaultId)
```
[the remark this answers is not in the document]
Comment (tqin7) [reply protocol/x/vault/keeper/grpc_query_shares.go:33]: nope ordering is not guaranteed. doesn't matter though as it's just a query

File: protocol/x/vault/keeper/grpc_query_shares_test.go (hunk 1 above)
Comment (tqin7) [reply protocol/x/vault/keeper/grpc_query_shares_test.go:75]: > paginates beyond max size

then I believe all vaults will be returned

yeah query is not guaranteed to have updated information

File: protocol/x/vault/keeper/grpc_query_shares.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Christopher-Li) [reply protocol/x/vault/keeper/grpc_query_shares.go:33]: If order is not guaranteed, wouldn't pagination feel?

**Review by Christopher-Li — APPROVED** [review]

File: protocol/x/vault/keeper/grpc_query_shares.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tqin7) [reply protocol/x/vault/keeper/grpc_query_shares.go:33]: hmmm good point let me look into this

**Comment by tqin7** [comment]
@mergifyio backport release/protocol/v5.x

**Pull request merged**